### PR TITLE
Fix warnings in ByValArray

### PR DESCRIPTION
**Pull request by yizhang82**

### CODE REVIEW

**Comment by yizhang82** [comment]
@dotnet-bot Test OSX x64 Checked Build and Test please

**Comment by yizhang82** [comment]
Fix #3451

File: tests/src/Interop/ArrayMarshalling/ByValArray/MarshalArrayByValNative.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ helper function
 LPSTR ToString(int i)
 {
     CHAR *pBuffer = (CHAR *)::CoTaskMemAlloc(10 * sizeof(CHAR)); // 10 is enough for our case, WCHAR for BSTR
-	snprintf(pBuffer, 10 * sizeof(CHAR), "%d", i);
+	snprintf(pBuffer, 10, "%d", i);
```
Comment (RussKeldorph) [inline tests/src/Interop/ArrayMarshalling/ByValArray/MarshalArrayByValNative.cpp]: Not enough indentation.

**Comment by RussKeldorph** [comment]
Does this address #4210 as well?

**Pull request merged**